src/YHttpRootHandler.cc

```diff
@@ -16,128 +16,28 @@
 
 #include "YHttpRootHandler.h"
 #include "YHttpServer.h"
+#include "YJsonSerializer.h"
 
 #include <microhttpd.h>
 #include <string>
 #include <boost/algorithm/string/replace.hpp>
+#include <json/json.h>
+
+const std::string YHttpRootHandler::documentation = "https://github.com/libyui/libyui-rest-api/#libyui-embedded-webserver";
 
 void YHttpRootHandler::body(struct MHD_Connection* connection,
     const char* url, const char* method, const char* upload_data,
     size_t* upload_data_size, std::ostream& body, bool *redraw)
 {
-    // TODO: save this to a standalone HTML file, editing HTML in a C string
-    // literal is cumbersome, use bin2c or something like that for converting
-    // to a C source file...
-    std::string help =
-"<html>"
-"  <head><title>LibYUI Embedded Webserver</title></head>"
-"  <body>"
-"    <h1>LibYUI Embedded Webserver</h1>"
-"    <p>This webserver provides a REST API for the LibYUI application.</p>"
-"      <p>It can be used for testing and controlling the application in automated tests.</p>"
-"      <br>"
-"      <h2>Short Documentation</h2>"
-"      <h3>Application</h3>"
-"      <p>Request: GET <a href='/application'>/application</a></p>"
-"      <h4>Description</h4>"
-"      <p>Get the application and UI generic properties like text or graphical mode,"
-"      dialog size, screen size and supported UI featues.</p>"
-"      <h4>Response</h4>"
-"      <p>JSON format</p>"
-"      <h4>Examples</h4>"
-"      <p>"
-"        <pre>  curl http://localhost:@port@/application</pre>"
-"      </p>"
-"      <hr>"
-"      <h3>Dump Whole Dialog</h3>"
-"      <p>Request: GET <a href='/dialog'>/dialog</a></p>"
```

Comment:
@lslezak The idea sounds reasonable, but unfortunately it requires additional efforts on maintenance as it should be duplicated both in README.md and on root (`/` endpoint).

Maybe I just do not know the initial plan on what is the expected use cases for the API. I guess here we should come to consensus on what is the less evil and try to define the target audience:) 

I as a test developer will probably hardly use the links as most of the time I'll send the requests using [Client](https://github.com/jknphy/libyui_client). Newcomers will be able to proceed by `documentation_url` link and also see the required endpoints in case they want to get familiar with the API itself.

Could you please let me know if you are OK with the proposed solution with the link to documentation in JSON, otherwise I'll return the HTML documentation.